src/H5Dmpio.c

```diff
@@ -3271,6 +3210,20 @@ H5D__mpio_redistribute_shared_chunks(H5D_filtered_collective_io_info_t *chunk_li
                                                  io_info, mpi_rank, mpi_size) < 0)
         HGOTO_ERROR(H5E_DATASET, H5E_CANTREDISTRIBUTE, FAIL, "can't redistribute shared chunks")
 
+    /*
+     * Now that chunks have been redistributed, each rank must update
+     * their chunk list struct's `num_chunks_to_read` field since it
+     * may now be out of date.
+     */
+    for (i = 0; i < chunk_list->num_chunk_infos; i++) {
```

Comment:
It looks like this is redundant since this scan is also performed in H5D__mpio_redistribute_shared_chunks_int(). Am I missing something?

Reply:
Thanks for the catch - I forgot to remove the old version after moving it down into the function